src/System.Text.Json/tests/JsonElementWriteTests.cs

```diff
@@ -901,19 +905,31 @@ public static void WritePropertyOutsideObject(bool skipValidation)
                     {
                         JsonTestHelper.AssertThrows<InvalidOperationException>(
                             ref writer,
-                            (ref Utf8JsonWriter w) => val.WriteProperty(CharLabel, w));
+                            (ref Utf8JsonWriter w) =>
+                            {
+                                w.WritePropertyName(CharLabel);
+                            });
 
                         JsonTestHelper.AssertThrows<InvalidOperationException>(
                             ref writer,
-                            (ref Utf8JsonWriter w) => val.WriteProperty(CharLabel.AsSpan(), w));
+                            (ref Utf8JsonWriter w) =>
+                            {
+                                w.WritePropertyName(CharLabel.AsSpan());
+                            });
 
                         JsonTestHelper.AssertThrows<InvalidOperationException>(
                             ref writer,
-                            (ref Utf8JsonWriter w) => val.WriteProperty(byteUtf8, w));
+                            (ref Utf8JsonWriter w) =>
+                            {
+                                w.WritePropertyName(byteUtf8);
+                            });
 
                         JsonTestHelper.AssertThrows<InvalidOperationException>(
                             ref writer,
-                            (ref Utf8JsonWriter w) => val.WriteProperty(JsonEncodedText.Encode(CharLabel), w));
+                            (ref Utf8JsonWriter w) =>
+                            {
+                                w.WritePropertyName(JsonEncodedText.Encode(CharLabel));
+                            });
                     }
 
                     writer.Flush();
```

Comment:
The test for sending a null writer to JsonElement.WriteTo should be in this file.